### PR TITLE
Various improvements to the 5.0 release notes

### PR DESCRIPTION
**Pull request by abock**
1. fix typos and formatting in many places
2. spruce up the C# 7 blurb
3. add `csharp` scripting improvement

### CODE REVIEW

File: docs/about-mono/releases/5.0.0.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ Highlights
 * [Enabling concurrent SGen garbage collector](#sgen-improvements) to reduce time spent in GC
 * [Introducing the AppleTLS stack](#appletls-stack) on macOS for HTTPS connections
 * [Continued Progress on .NET Class Library convergence](#corefx--reference-source-adoption)
-* [Updated libjpg in MacOS package](#libjpg-update-in-macos-package)
+* [Updated libjepg in macOS package](#libjpeg-update-in-macos-package)
```
Comment (lextm) [inline docs/about-mono/releases/5.0.0.md]: Should be libjpeg, not libjepg.

Comment (abock) [reply docs/about-mono/releases/5.0.0.md]: Heh, you try to fix a typo and you introduce another.

Comment (abock) [reply docs/about-mono/releases/5.0.0.md]: Fixed.

**Comment by abock** [comment]
Merging at @migueldeicaza's request.

**Pull request merged**